ground/src/main/java/com/google/android/ground/repository/OfflineAreaRepository.kt

```diff
@@ -203,4 +209,49 @@ constructor(
           .andThen(localTileSetStore.deleteTileSetByUrl(tileSet))
       }
       .andThen(localOfflineAreaStore.deleteOfflineArea(offlineAreaId))
+
+  /**
+   * Downloads tiles in the specified bounds and stores them in the local filesystem. Emits the
+   * number of bytes processed and total expected bytes as the download progresses.
+   */
+  suspend fun downloadTiles(bounds: Bounds): Flow<Pair<Int, Int>> = flow {
+    val client = getMogClient()
+    val requests = client.buildTilesRequests(bounds.toGoogleMapsObject())
+    val totalBytes = requests.sumOf { it.totalBytes }
+    var bytesDownloaded = 0
+    val tilePath = getLocalTileSourcePath()
+    MogTileDownloader(client, tilePath).downloadTiles(requests).collect {
+      bytesDownloaded += it
+      emit(Pair(bytesDownloaded, totalBytes))
+    }
+  }
+
+  fun getLocalTileSourcePath(): String {
+    // TODO(#1730): Generate local tiles path based on source base path.
+    return File(fileUtil.filesDir.path, "tiles").path
+  }
+  /**
+   * Uses the first tile source URL of the currently active survey and returns a [MogClient], or
+   * throws an error if no survey is active or if no tile sources are defined.
+   */
+  private suspend fun getMogClient(): MogClient {
+    // TODO(#1730): Make sub-paths configurable and stop hardcoding here.
+    val baseUrl = getFirstTileSourceUrl()
+    val mogCollection =
+      MogCollection(
+        listOf(MogSource("${baseUrl}/world.tif", 0..7), MogSource("${baseUrl}/{x}/{y}.tif", 8..14))
+      )
+    // TODO(#1754): Create a factory and inject rather than instantiating here. Add tests.
+    return MogClient(mogCollection)
+  }
+
+  /**
+   * Returns the URL of the first tile source in the current survey, or throws an error if no survey
+   * is active or if no tile sources are defined.
+   */
+  private suspend fun getFirstTileSourceUrl(): String {
+    val survey = surveyRepository.activeSurveyFlowable.asFlow().firstOrNull()?.orElse(null)
```

Comment:
Is there any particular reason for using `.activeSurveyFlowable` instead of `.activeSurvey`?

Reply:
Good catch, resolved.